grammar.js

```diff
@@ -722,7 +725,7 @@ const rules = {
       seq(
         $._variablish,
         field('selection_operator', choice('?->', '->')),
-        $._variablish,
+        choice($._variablish, $.embedded_brace_expression, alias($._keyword, $.identifier)),
```

Comment:
Tested this out in Hacklang and even function calls are allowed in that brace. The Hacklang parser calls this node a `braced_expression` both in selection expressions and xhp expressions.

```
$value->{ $name }; // braced_expression
<frag>{ $name }</frag>; // braced_expression
```

So I think we can just use `xhp_braced_expression` here but rename it to `braced_expression`.

Reply:
Sounds good! I didn't want to make the additions too permissive at first, but it looks like Hack allows more than expected. Ex. `$dynamic_item->{new $fun->test()};` does not produce hh errors despite its weirdness.